test/functional/apps/context/_discover_navigation.ts

```diff
@@ -48,6 +48,7 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
       for (const [columnName, value] of TEST_FILTER_COLUMN_NAMES) {
         await filterBar.addFilter({ field: columnName, operation: 'is', value });
+        await PageObjects.header.waitUntilLoadingHasFinished();
```

Comment:
Likely wouldn't make much of a difference in this context since the loading is probably fast anyway, but I wonder if it would have worked to wait for loading just once after the for loop instead.

Reply:
@davismcphee This implementation ensures that the runner is waiting for the right loading (not the loading which was triggered by a previous action). I would like to keep it this way to avoid any other flakiness.